main.py

```diff
@@ -19,144 +31,94 @@ def get_spell_details(spell_url):
-        # If damage_at_slot_level is a string, create a list with a dictionary
-        damage_at_slot_level = [{'level': 'N/A', 'damage': damage_at_slot_level}]
-    elif not isinstance(damage_at_slot_level, list):
-        # If damage_at_slot_level is neither a string nor a list, create an empty list
-        damage_at_slot_level = []
-    damage_table = (
-            '\n| Slot Level | Damage |\n|------------|--------|\n' +
-            '\n'.join([f'| {level.get("level", "N/A")} | {level.get("damage", "N/A")} |' for level in damage_at_slot_level])
-    )
-    # Schools, Classes, Subclasses with backlinks
-    school = spell_data.get('school',{}).get('name','')
-    classes = '  '.join([f'[[{c["name"]}]]' for c in spell_data.get('classes',[])])
-    subclasses = '  '.join([f'[[{sub_c["name"]}]]' for sub_c in spell_data.get('subclasses',[])])
-    #description+++ = content+++
-    content = f'# {spell_name}\n\n'
-    content += f'{spell_desc}\n\n'  # Description as the main content
-    content += f'## Higher Level\n{higher_level}\n\n'
-    content += f'## Details\n'
-    content += f'- Range: {range_val}\n'
-    content += f'- Components: {components}\n'
-    content += f'- Material: {material}\n'
-    content += f'- Ritual: {ritual}\n'
-    content += f'- Duration: {duration}\n'
-    content += f'- Concentration: {concentration}\n'
-    content += f'- Casting Time: {casting_time}\n'
-    content += f'- Attack Type: {attack_type}\n'
-    content += f'- Damage Type: {damage_type}\n'
-    content += damage_table
-    #backlinks
-    content += f'\n- School: [[{school}]]\n'
-    content += f'- Classes: {classes}\n'
-    content += f'- Subclasses: {subclasses}\n'
-    return content
-
-def save_spell_md_file(spell_details, content, save_directory):
-    spell_name = spell_details.get('name', 'Unknown Spell')
-    file_path = os.path.join(save_directory, f'{spell_name}.md')
+
+def save_spell_md_file(spell_details, spell_template, save_directory):
+    spell_name = spell_details['name']
+    # Replace '/' with another character (curse you antipathy/sympathy, blindness/deafness) in the spell name
+    sanitized_spell_name = spell_name.replace('/', '_').title()
+    file_path = os.path.join(save_directory, f'{sanitized_spell_name}.md')
+    content = spell_template.render(spell_data=spell_details)
+
     if os.path.exists(file_path):
         user_response = input(f'The file {file_path} already exists. Do you want to overwrite it? (y/n): ')
         if user_response.lower() != 'y':
             print(f'Skipping {file_path}.')
             return False  # File was not saved
+
     with open(file_path, 'w', encoding='utf-8') as file:
         file.write(content)
+
     print(f'Created file: {file_path}')
     return True  # File was saved successfully
 
+
+def saving_result(spell_details, spell_template, save_directory):
+    if spell_details:
+        # Ensure that the save directory exists
+        if not ensure_valid_directory(save_directory):
```

Comment:
the save_directory is defined once and it is static. 
this means that you do not need to make this check for every spell but only once at the start of the program itself.